### PR TITLE
docs: Update CREATEROLE attribute docs

### PR DESCRIPTION
**Pull request by jkosh44**
### Motivation
Update docs.

### Checklist

- [X] This PR has adequate test coverage / QA involvement has been duly considered.
- [X] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Comment by jkosh44** [comment]
I modeled this after the PostgreSQL language, which is similar. From https://www.postgresql.org/docs/current/role-attributes.html

> Because the `CREATEROLE` privilege allows a user to grant or revoke membership even in roles to which it does not (yet) have any access, a `CREATEROLE` user can obtain access to the capabilities of every predefined role in the system, including highly privileged roles such as `pg_execute_server_program` and `pg_write_server_files`.

**Comment by jkosh44** [comment]
FYI @tr0njavolta

File: doc/user/content/sql/alter-role.md [hunk 1]
```diff
@@ -23,8 +23,8 @@ Field               | Use
 --------------------|-------------------------------------------------------------------------
 _role_name_         | A name for the role.
 **INHERIT**         | Grants the role the ability to inheritance of privileges of other roles.
-**CREATEROLE**      | Grants the role the ability to create, alter, and delete roles.
-**NOCREATEROLE**    | Denies the role the ability to create, alter, and delete roles.
+**CREATEROLE**      | Grants the role the ability to create, alter, delete roles and the ability to grant and revoke role membership. This attribute is very powerful. It allows roles to grant and revoke membership in other roles, even if it doesn't have explicit membership in those roles. As a consequence, any role with this attribute can obtain the privileges of any other role in the system.
+**NOCREATEROLE**    | Denies the role the ability to create, alter, delete roles and the ability to grant and revoke role membership.
```
Comment (tr0njavolta) [inline doc/user/content/sql/alter-role.md:27]:
```suggestion
**NOCREATEROLE**    | Denies the role the ability to create, alter, delete roles or grant and revoke role membership.
```

File: doc/user/content/sql/create-role.md [hunk 2]
```diff
@@ -26,8 +26,8 @@ Field               | Use
 --------------------|-------------------------------------------------------------------------
 _role_name_         | A name for the role.
 **INHERIT**         | Grants the role the ability to inheritance of privileges of other roles.
-**CREATEROLE**      | Grants the role the ability to create, alter, and delete roles.
-**NOCREATEROLE**    | Denies the role the ability to create, alter, and delete roles.
+**CREATEROLE**      | Grants the role the ability to create, alter, delete roles and the ability to grant and revoke role membership. This attribute is very powerful. It allows roles to grant and revoke membership in other roles, even if it doesn't have explicit membership in those roles. As a consequence, any role with this attribute can obtain the privileges of any other role in the system.
+**NOCREATEROLE**    | Denies the role the ability to create, alter, delete roles and the ability to grant and revoke role membership.
```
Comment (tr0njavolta) [inline doc/user/content/sql/create-role.md:30]:
```suggestion
**NOCREATEROLE**    | Denies the role the ability to create, alter, delete roles or grant and revoke role membership.
```

**Review by tr0njavolta — APPROVED** [review]
Small wording change for clarity.

**Pull request merged**